nautobot/extras/tests/test_api.py

```diff
@@ -661,6 +661,28 @@ def test_run_git_sync_with_permissions(self, mock_get_worker_count):
         response = self.client.post(url, format="json", **self.header)
         self.assertHttpStatus(response, status.HTTP_200_OK)
 
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=[])
+    @skipIf(
+        "dummy_plugin" not in settings.PLUGINS,
+        "dummy_plugin not in settings.PLUGINS",
+    )
+    def test_create_with_plugin_provided_contents(self):
+        """Test that `provided_contents` published by a plugin works."""
+        self.add_permissions("extras.add_gitrepository")
+        self.add_permissions("extras.change_gitrepository")
+        url = self._get_list_url()
+        data = {
+            "name": "plugin_test",
+            "slug": "plugin-test",
+            "remote_url": "https://localhost/plugin-test",
+            "provided_contents": ["dummy_plugin.textfile"],
+        }
+        response = self.client.post(url, data, format="json", **self.header)
+        self.assertHttpStatus(response, status.HTTP_201_CREATED)
+        # FIXME(jathan): Figure out why response.data is coming back as a set and requires being
+        # cast to a list??
```

Comment:
That's standard behavior of a DRF MultipleChoiceField, I think. Note that our `MultipleChoiceJSONField` overrides `to_internal_value` to return a sorted list rather than a set, but does *not* override `to_representation` so that will still return a set... I guess we could consider overriding that method too, but not sure if that's desirable or what side effects it might have.

Reply:
Yeah I see. I think this is fine for now.